apps/backend/app/core/database.py

```
@@ -117,16 +212,81 @@ def get_sync_db_session() -> Generator[Session, None, None]:
     async with async_engine.begin() as conn:
-        await conn.run_sync(Base.metadata.create_all)
+        await conn.run_sync(base.metadata.create_all)
+
+
+async def check_database_connection() -> bool:
+    """Check if database is accessible."""
+    try:
+        async with async_engine.connect() as conn:
+            await conn.execute(text("SELECT 1"))
+        return True
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)